internal/store/segment/api.go

```diff
@@ -122,10 +124,91 @@ func (s *segmentServer) AppendToBlock(
 	if err != nil {
 		return nil, err
 	}
-
 	return &segpb.AppendToBlockResponse{Offsets: offs}, nil
 }
 
+func (s *segmentServer) AppendToBlockStream(stream segpb.SegmentServer_AppendToBlockStreamServer) error {
+	ctx := context.Background()
+	for {
+		request, err := stream.Recv()
+		if err != nil {
+			log.Warning(ctx, "append stream recv failed", map[string]interface{}{
+				log.KeyError: err,
+			})
+			return err
+		}
+
+		go func(stream segpb.SegmentServer_AppendToBlockStreamServer, req *segpb.AppendToBlockStreamRequest) {
```

Comment:
Use `chan` to async process `AppendToBlockStreamRequest`, Don't create goroutine for each request

Reply:
and should use goroutine pool